README.md
# sliding_images

`SlidingImages` Flutter widget creates image sliders with customizable properties such as image dimensions, padding, auto-play, custom indicators, and more..

![demo](/demo.gif)

## Features

 - Image carousel with URL support.
 - Customizable image dimensions and padding.
 - Set the initial displayed image index.
 - Auto-play with interval.
 - Apply border to images.
 - Navigation indicator.
 - Animation curve customization.
 - Scaling for active and non-active images.

## Getting started

To use this package, add sliding_images as a dependency in your pubspec.yaml file.

## Usage

Minimal example:

```dart
SlidingImages(
  imgUrls: ["https://p1.pxfuel.com/preview/321/155/730/alpine-bavaria-flowers-germany.jpg",
            "https://p1.pxfuel.com/preview/818/670/574/dogs-chihuahua-rest-small-fur-cozy.jpg",
            "https://p0.pxfuel.com/preview/961/674/322/podcast-coffee-chocolate-cup.jpg",
            ],
)

```

Custom settings:

```dart
    SlidingImages(
      imgUrls: ["https://p1.pxfuel.com/preview/321/155/730/alpine-bavaria-flowers-germany.jpg",
                "https://p1.pxfuel.com/preview/818/670/574/dogs-chihuahua-rest-small-fur-cozy.jpg",
                "https://p0.pxfuel.com/preview/961/674/322/podcast-coffee-chocolate-cup.jpg",
                ],
      imageWidth: 300,
      imageHeight: 200,
      padding: EdgeInsets.all(8.0),
      border: Border.all(color: Colors.grey, width: 1.0),
      autoPlayInterval: Duration(seconds: 3),
      customIndicator: CustomIndicator(),
      initialPage: 2,
      enableAutoPlay: true,
      infiniteScroll: true,
      animationCurve: Curves.easeInOut,
      activeSlideScale: 1.2,
      nonActiveSlideScale: 0.9,
    )
```

## See also

 - [github repo](https://github.com/jyotsnayall/sliding_images-flutter-plugin)
 - [pub.dev package](https://pub.dev/packages/sliding_images)


CHANGELOG.md
## 0.0.3
* change plugin description

## 0.0.2

## 0.0.1

* Initial release.


test/sliding_images_flutter_plugin_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:sliding_images_flutter_plugin/sliding_images_flutter_plugin.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


lib/demo.dart
library sliding_images;

import 'package:carousel_slider/carousel_slider.dart';
import 'package:flutter/widgets.dart';

class SlidingImages extends StatefulWidget {
  final double? imageWidth;
  final double? imageHeight;
  final List<String> imgUrls;
  final EdgeInsetsGeometry? padding;
  final Border? border;
  final Duration autoPlayInterval;
  final Widget? customIndicator;
  final int initialPage;
  final bool enableAutoPlay;
  final bool infiniteScroll;
  final Curve animationCurve;
  final double activeSlideScale;
  final double nonActiveSlideScale;

  const SlidingImages({
    Key? key,
    this.imageWidth,
    this.imageHeight,
    required this.imgUrls,
    this.padding,
    this.border,
    this.autoPlayInterval = const Duration(seconds: 5),
    this.customIndicator,
    this.initialPage = 0,
    this.enableAutoPlay = true,
    this.infiniteScroll = true,
    this.animationCurve = Curves.easeInOut,
    this.activeSlideScale = 1.0,
    this.nonActiveSlideScale = 0.8,
  }) : super(key: key);

  @override
  State<SlidingImages> createState() => _SlidingImagesState();
}

class _SlidingImagesState extends State<SlidingImages> {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: CarouselSlider.builder(
        itemCount: widget.imgUrls.length,
        itemBuilder: (context, index, pageViewIndex) {
          return Container(
            width: widget.imageWidth ?? 300,
            height: widget.imageHeight ?? 300,
            padding: widget.padding ?? EdgeInsets.all(2.0),
            decoration: BoxDecoration(
              border: widget.border,
            ),
            child: Image.network(
              widget.imgUrls.elementAt(index),
              fit: BoxFit.cover,
            ),
          );
        },
        options: CarouselOptions(
          autoPlay: widget.enableAutoPlay,
          autoPlayInterval: widget.autoPlayInterval,
          enlargeCenterPage: true,
          initialPage: widget.initialPage,
          enableInfiniteScroll: widget.infiniteScroll,
          scrollDirection: Axis.horizontal,
          scrollPhysics: ClampingScrollPhysics(),
          pageSnapping: true,
          reverse: false,
          pauseAutoPlayOnTouch: true,
          autoPlayCurve: widget.animationCurve,
          disableCenter: false,
        ),
      ),
    );
  }
}


lib/sliding_images.dart
library sliding_images;

import 'package:carousel_slider/carousel_slider.dart';
import 'package:flutter/widgets.dart';

class SlidingImages extends StatefulWidget {
  final double? imageWidth;
  final double? imageHeight;
  final List<String> imgUrls;
  final EdgeInsetsGeometry? padding;
  final Border? border;
  final Duration autoPlayInterval;
  final Widget? customIndicator;
  final int initialPage;
  final bool enableAutoPlay;
  final bool infiniteScroll;
  final Curve animationCurve;
  final double activeSlideScale;
  final double nonActiveSlideScale;

  const SlidingImages({
    Key? key,
    this.imageWidth,
    this.imageHeight,
    required this.imgUrls,
    this.padding,
    this.border,
    this.autoPlayInterval = const Duration(seconds: 5),
    this.customIndicator,
    this.initialPage = 0,
    this.enableAutoPlay = true,
    this.infiniteScroll = true,
    this.animationCurve = Curves.easeInOut,
    this.activeSlideScale = 1.0,
    this.nonActiveSlideScale = 0.8,
  }) : super(key: key);

  @override
  State<SlidingImages> createState() => _SlidingImagesState();
}

class _SlidingImagesState extends State<SlidingImages> {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: CarouselSlider.builder(
        itemCount: widget.imgUrls.length,
        itemBuilder: (context, index, pageViewIndex) {
          return Container(
            width: widget.imageWidth ?? 300,
            height: widget.imageHeight ?? 300,
            padding: widget.padding ?? const EdgeInsets.all(2.0),
            decoration: BoxDecoration(
              border: widget.border,
            ),
            child: Image.network(
              widget.imgUrls.elementAt(index),
              fit: BoxFit.cover,
            ),
          );
        },
        options: CarouselOptions(
          autoPlay: widget.enableAutoPlay,
          autoPlayInterval: widget.autoPlayInterval,
          enlargeCenterPage: true,
          initialPage: widget.initialPage,
          enableInfiniteScroll: widget.infiniteScroll,
          scrollDirection: Axis.horizontal,
          scrollPhysics: ClampingScrollPhysics(),
          pageSnapping: true,
          reverse: false,
          pauseAutoPlayOnTouch: true,
          autoPlayCurve: widget.animationCurve,
          disableCenter: false,
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:sliding_images/sliding_images.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  List<String> imgUrls = [
    "https://p1.pxfuel.com/preview/321/155/730/alpine-bavaria-flowers-germany.jpg",
    "https://p1.pxfuel.com/preview/818/670/574/dogs-chihuahua-rest-small-fur-cozy.jpg",
    "https://p0.pxfuel.com/preview/961/674/322/podcast-coffee-chocolate-cup.jpg",
    "https://p1.pxfuel.com/preview/497/593/209/coffee-winter-warmth-cozy.jpg",
    "https://p1.pxfuel.com/preview/261/554/537/cat-wooden-bench-animal-nature.jpg",
    "https://p1.pxfuel.com/preview/806/13/221/budapest-chain-bridge-danube-hungary-royalty-free-thumbnail.jpg"
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: const Text("Plugin Example"),
      ),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              "Minimal Features Example:",
              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            SlidingImages(
              imgUrls: imgUrls,
            ),
            SizedBox(height: 20),
            Text(
              "Customized Example 1: Different Image Dimensions & Padding",
              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            SlidingImages(
              imgUrls: imgUrls,
              imageWidth: 250,
              imageHeight: 150,
              padding: EdgeInsets.all(10.0),
            ),
            SizedBox(height: 20),
            Text(
              "Customized Example 2: Auto-play with Custom Indicator",
              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            SlidingImages(
              imgUrls: imgUrls,
              enableAutoPlay: true,
              autoPlayInterval: Duration(seconds: 4),
              customIndicator: CustomIndicator(),
            ),
            SizedBox(height: 20),
            Text(
              "Customized Example 3: Border & Scaling Effects",
              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            SlidingImages(
              imgUrls: imgUrls,
              imageWidth: 350,
              imageHeight: 250,
              border: Border.all(color: Colors.grey, width: 2.0),
              activeSlideScale: 1.1,
              nonActiveSlideScale: 0.9,
            ),
          ],
        ),
      ),
    );
  }
}

class CustomIndicator extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    // Implement your custom indicator widget
    return Container(
      // Your custom indicator UI here
      width: 10,
      height: 10,
      margin: EdgeInsets.symmetric(horizontal: 2),
      decoration: BoxDecoration(
        shape: BoxShape.circle,
        color: Colors.blue,
      ),
    );
  }
}